docs/auth/identity-resolver.md

```diff
@@ -90,116 +83,156 @@ the given auth provider, as well as a context object that contains various helpe
 for looking up users and issuing tokens. There are also a number of built-in sign-in
 resolvers that can be used, which are covered a bit further down.
 
-Note that while it possible to configure multiple auth providers to be used for sign-in,
-you should take care when doing so. It is best to make sure that the different auth
-providers either do not have any user overlap, or that any users that are able to log
-in with multiple providers always end up with the same Backstage identity.
+Note that while it possible to configure multiple auth providers to be used for
+sign-in, you should take care when doing so. It is best to make sure that the
+different auth providers either do not have any user overlap, or that any users
+that are able to log in with multiple providers always end up with the same
+Backstage identity. For most organizations, it makes the most sense to provide
+only one sign-in method.
+
+### Using Builtin Resolvers
+
+Most auth providers come with a set of builtin sign in providers that you can
+choose from. They target the most common use cases, and if they fit your needs,
+you can pick one or more of them without having to write any code at all. You
+still have to make a choice - as mentioned above, even if there are a set of
+builtins, none of them are selected by default.
+
+You set up builtin sign in resolvers using [your app-config](../conf/index.md),
+next to the respective provider's configuration. Here's an example for Microsoft
+Azure:
+
+```yaml title="in e.g. app-config.yaml"
+auth:
+  environment: development
+  providers:
+    microsoft:
+      development:
+        clientId: ${AZURE_CLIENT_ID}
+        clientSecret: ${AZURE_CLIENT_SECRET}
+        tenantId: ${AZURE_TENANT_ID}
+        signIn:
+          resolvers:
+            - resolver: emailMatchingUserEntityAnnotation
```

Comment:
Where do we define these resolver's names to be used in config? If I create a custom provider, can I select resolver via config as well? It is a question that came to my mind while reading the docs but I couldn't find the answer...

Reply:
It's either/or - you provide code, or you provide the config way.

It works such that there is a default resolver implementation that reads those config flags. If you provide one in code, you replace that implementation.

the list of (named) sign in resolvers that your provider supports through config, is decided by its provider factory https://github.com/backstage/backstage/blob/42770dab28085ba2a5696b3b73251a5ff8ffd414/plugins/auth-backend-module-github-provider/src/module.ts#L39